vendor/bat-native-ledger/src/bat_helper.cc

```diff
@@ -2618,6 +2619,25 @@ uint64_t getMediaDuration(const std::map<std::string, std::string>& data,
   return duration;
 }
 
+// static
+bool isValidLeadingUrlSubString(
+    const std::string& url_to_validate,
+    const std::string& to_match) {
+  GURL gurl(url_to_validate);
+  const std::string host_portion =
+    to_match.substr(
+      to_match.find("://") != std::string::npos ? to_match.find("://") + 3 : 0,
+      to_match.find("/"));
+  return gurl.is_valid() &&
+         gurl.spec().find(to_match) != std::string::npos &&
+         gurl.host().find(host_portion) != std::string::npos &&
+         !to_match.empty();
+}
+
+bool isValidUrl(const std::string& url_to_validate) {
+  return GURL(url_to_validate).is_valid();
```

Comment:
this method isn't necessary because `is_valid` needs to be checked inside `isValidLeadingUrlSubString`

Reply:
Working a separate case where we can have a partial `.example.com` or the full `https://example.com` (no leading `.`)

Reply:
but it already validates in `isValidLeadingUrlSubString` so what is the point of the extra call afterwards? If anything we should be checking before. Either way I think we should dump this method and call `GURL(...).is_valid()` directly because the tests for `IsValidUrl` are just testing GURL::is_valid which already has plenty of tests

Reply:
Agree. This was doing something other than what I intended. Was able to compensate in `IsValidLeadingUrlSubString` and removed this.